src/api/app/models/group.rb

```diff
@@ -203,9 +203,9 @@ def maintainer_roler
     @maintainer_roler ||= Role.hashed['maintainer']
   end
 
-  def delete_user(klass, login_id, group_id)
+  def delete_user(klass, login_id, group_id, user_session_login: nil)
```

Comment:
Strictly speaking who deletes a user can be anyone, not just the user session. I would rename `user_session_login` to just `user`

Reply:
> Strictly speaking who deletes a user can be anyone, not just the user session. I would rename `user_session_login` to just `user`

I don't agree. That "who deletes a user can be anyone" is not correct. `delete_user` is only called by two methods, inside controllers, with an authenticated user:
- `remove_user`, called by `app/controllers/webui/groups/users_controller.rb` and `app/controllers/group_controller.rb`, and
- `update_from_xml`, called by `app/controllers/group_controller.rb`.

The variable name `user_session_login` also has the advantage of making it clearer the difference with other variables: `id`, `login_id`, `group_id`.

Reply:
Well, yes, but the method doesn't and shouldn't care if that user is coming from the session or not, there is nothing that requires that. So from the method's perspective this is only to telling who deleted the user, nothing else.

Reply:
There we agree: the method doesn't care if that user is coming from the session or not.

Reply:
In other words, I would prefer to keep the variable name like I proposed, `user_session_login`. Changing it to any other name would lead to more confusion than clarity, IMHO.

Reply:
Alright